4/AlgoritmLesson4/TreeSearch.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AlgoritmLesson4
+{
+   public class TreeSearch : Itree
+    {
+        public int Depth { get; set; }
+        public TreeNode Node { get; set; }
+        public TreeNode Root { get; set; }
+
+        public TreeNode root;
+
+
+        public void AddItem(int value)//добавить узел
+        {
+            
+            var newnode = new TreeNode {Value = value };
+            if (root == null)
+            {
+                root = newnode;
+            }
+            else
+            {
+                TreeNode temp = root;
+                TreeNode parent = null;
+
+                while(temp != null)
+                {
+                    parent = temp;
+                    if(value < temp.Value)
+                    {
+                        temp = temp.LeftChild;
+                        if(temp == null)
+                        {
+                            parent.LeftChild = newnode;
+                        }
+                    }
+                    else
+                    {
+                        temp = temp.RightChild;
+                        if (temp == null)
+                        {
+                            parent.RightChild = newnode;
+                        }
+                    }
+                }
+            }
+           
+        }
+
+        public void PrintTree()
+        {
+            root.Print();
+        }
+
+        public TreeNode GetNodeByValue(int value)
+        {
+            if( root != null)
+            {
+                TreeNode findnode = root;
+
+                while(findnode != null)
+                {
+                    if(value == findnode.Value)
+                    {
+                        return findnode;
+                    }
+                    else if(value > findnode.Value)
+                    {
+                        findnode = findnode.RightChild;
+                    }
+                    else
+                    {
+                        findnode = findnode.LeftChild;
+                    }
+                }
+                return null;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public TreeNode GetRoot() => root;//получение корневого узла
+        
+        public TreeNode takeancestor (TreeNode delnode)
```

Comment:
Название метода должно быть с большой буквы. Входная переменная - тоже.
```suggestion
        public TreeNode TakeAncestor (TreeNode DelNode)
```